app/src/lib/tokens/Word.svelte

```diff
@@ -1,18 +1,15 @@
 <script>
-	import {TOKEN_TYPE, token_has_concept} from '$lib/parser/token'
-	import Error from './Error.svelte'
+	import {token_has_concept} from '$lib/parser/token'
 	import NotFound from './NotFound.svelte'
 	import Result from './Result.svelte'
 
 	/** @type {Token} */
 	export let token
-	export let classes = ''
+	export let should_join = false
 </script>
 
-{#if token.type === TOKEN_TYPE.ERROR || token.message.length > 0}
-	<Error {token} {classes} />
-{:else if token_has_concept(token)}
-	<Result {token} {classes} />
+{#if token_has_concept(token)}
+	<Result {token} {should_join} />
```

Comment:
if my feedback above is true, I don't think you need `{should_join}` at all here or below.